src/main/asciidoc/jdbc.adoc

```diff
@@ -86,108 +84,115 @@ class ApplicationConfig {
 You then have to register the custom implementation of `JdbcRepositoryFactory` as a Spring bean.
-Note that this should be rarely necessary. See the link:$$http://docs.spring.io/spring-data/data-jdbc/docs/current/api/index.html?org/springframework/data/jdbc/repository/support/JdbcRepositoryFactory.html$$[JavaDoc] for details.
+Note that this should rarely be necessary. See the link:$$http://docs.spring.io/spring-data/data-jdbc/docs/current/api/index.html?org/springframework/data/jdbc/repository/support/JdbcRepositoryFactory.html$$[JavaDoc] for details.
 |===============
 
 [[jdbc.entity-persistence.id-generation]]
-=== Id generation
+=== ID Generation
 
-Spring Data JDBC uses the id to identify entities.
-The id of an entity must be annotated with Spring Data's https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/annotation/Id.html[`@Id`] annotation.
+Spring Data JDBC uses the ID to identify entities.
+The ID of an entity must be annotated with Spring Data's https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/annotation/Id.html[`@Id`] annotation.
 
-When your data base has some autoincrement column for the id column the generated value will get set in the entity after inserting it into the database.
+When your data base has an auto-increment column for the ID column, the generated value gets set in the entity after inserting it into the database.
 
-One important constraint is that after saving an entity the entity must not be _new_ anymore.
-With autoincrement columns this happens automatically since the the id gets set by Spring Data with the value from the id column.
-If you are not using autoincrement columns you can use that using a `BeforeSave`-listener which sets the id of the entity (see below).
+One important constraint is that, after saving an entity, the entity must not be new any more.
+With auto-increment columns, this happens automatically, because the ID gets set by Spring Data with the value from the `id` column.
+If you are not using auto-increment columns, you can use a `BeforeSave` listener, which sets the `id` of the entity (covered later in this document).
 
 [[jdbc.query-methods]]
-== Query methods
+== Query Methods
+
+This section offers some specific information about the implementation and use of Spring Data JDBC.
 
 [[jdbc.query-methods.strategies]]
-=== Query lookup strategies
+=== Query Lookup Strategies
 
-The JDBC module only supports defining a query manually as a String in a `@Query` annotation.
+The JDBC module supports defining a query manually only as a String in a `@Query` annotation.
 Deriving a query from the name of the method is currently not supported.
 
 [[jdbc.query-methods.at-query]]
-=== Using @Query
+=== Using `@Query`
```

Comment:
Is the rule no code formatting in headings?

Reply:
I put code formatting in headings. I have observed that it works correctly in cross-references and the ToC, and I want the consistency with the body. (I gather that Asciidoc used to have trouble with it, but now it works.)